### PR TITLE
CIDEVSTC-43	 Implement data process management service extensions 

### PR DESCRIPTION
**Pull request by MauriceManning**
add transform worker lazy load of data process information
requires PR 482 merged first

@lukecampbell

### CODE REVIEW

File: ion/processes/data/transforms/test/test_transform_worker.py [hunk 1]
```diff
@@ -130,7 +131,6 @@ def create_data_processes(self):
             function='add_arrays',
             module="ion_example.add_arrays",
             arguments=['arr1', 'arr2'],
-            uri='http://sddevrepo.oceanobservatories.org/releases/ion_example-0.1-py2.7.egg' ,
```
Comment (lukecampbell) [inline ion/processes/data/transforms/test/test_transform_worker.py:133]: Why'd we remove the egg URI?

File: ion/processes/data/transforms/transform_worker.py [hunk 2]
```diff
@@ -126,100 +124,124 @@ def recv_packet(self, msg, stream_route, stream_id):
...
-                #load the details of this data process
-                dataprocess_info = self._dataprocesses[dp_id]
-                #todo: load once into a 'set' of modules?
-                #load the associated transform funcation
-                # Try to load the module, if it fails load the egg then try again
-                try:
-                    module = importlib.import_module(dataprocess_info.get_safe('module', '') )
-                except ImportError:
-                    # Download and install the egg
-                    egg = self.download_egg(dataprocess_info.get_safe('uri',''))
```
Comment (lukecampbell) [inline ion/processes/data/transforms/transform_worker.py:144]: I'm guessing this is why.

File: ion/processes/data/transforms/transform_worker.py [hunk 3]
```diff
@@ -126,100 +124,124 @@ def recv_packet(self, msg, stream_route, stream_id):
...
+        dataprocess_info = self._dataprocesses[dataprocess_id]
+        try:
+            #todo: load once into a 'set' of modules?
+            #load the associated transform funcation
+            module = importlib.import_module(dataprocess_info.get_safe('module', '') )
+            function = getattr(module, dataprocess_info.get_safe('function','') )
+            arguments = dataprocess_info.get_safe('arguments', '')
+            argument_list = dataprocess_info.get_safe('argument_map', {})
+        except ImportError:
+            log.error('Error running transform')
```
Comment (lukecampbell) [inline ion/processes/data/transforms/transform_worker.py:190]: You can throw that egg load code in here

``` python
egg = self.download_egg(dataprocess_info.get_safe('uri',''))
pkg_resources.working_set.add_entry(egg)
module = importlib.import_module(dataprocess_info.get_safe('module', '') )
```

File: ion/services/sa/process/data_process_management_service.py [hunk 4]
```diff
@@ -643,8 +685,8 @@ def _manage_attachments(self):
         pass
 
     def _create_subscription(self, dproc, in_data_product_ids):
-        stream_ids = [self._get_stream_from_dp(i) for i in in_data_product_ids]
-        #@TODO Maybe associate a data process with an exchange point but in the mean time: 
+        stream_ids = [self._get_stream_from_dataproduct(i) for i in in_data_product_ids]
+        #@TODO Maybe associate a data process with an exchange point but in the mean time:
         queue_name = 'sub_%s' % dproc.name
         subscription_id = self.clients.pubsub_management.create_subscription(name=queue_name, stream_ids=stream_ids)
```
Comment (lukecampbell) [inline ion/services/sa/process/data_process_management_service.py:690]: I think this should be multiple subscriptions, one for each data product

``` python
for data_product_id in in_data_product_ids:
    stream_id = self._get_stream_from_dp(data_product_id)
    subscription_id = self.clients.pubsub_management.create_subscription(name=queue_name, stream_ids=[stream_id])
```

File: ion/processes/data/transforms/transform_worker.py (hunk 2 above)
Comment (MauriceManning) [inline ion/processes/data/transforms/transform_worker.py:144]: yes, thought that was the better way to load the function. Not correct?

**Pull request merged**